### ISSUE TITLE
"host" header being incorrectly set in fetch_from_origin

### ISSUE THREAD

**Issue by tomdmaguire** [pintsized/ledge#98]
Hi there,

I am experiencing an issue in that ledge doesn't seem to correctly set a host header before using lua-resty-http to fetch from origin. When ledge grabs the nginx headers [here](https://github.com/pintsized/ledge/blob/master/lib/ledge/ledge.lua#L2077) in my test scenario the headers are as follows:

```
{
  connection = "close",
  ["content-length"] = "23",
  host = "localhost"
}
```

When these headers are then passed onto httpc:request [here](https://github.com/pintsized/ledge/blob/master/lib/ledge/ledge.lua#L2102), the http module then tries to request content from localhost, instead of ```upstream_host```.

I can solve this by explicitly setting the host header just after [this](https://github.com/pintsized/ledge/blob/master/lib/ledge/ledge.lua#L2077-L2079) as so:

```
headers['host'] = self:config_get("upstream_host")
```

But I feel like I am missing something.

Any direction would be greatly appreciated. Thanks

**Comment by forkfork** [comment]
@tomdmaguire would the PR #99 help with a slightly nicer approach for this one? Example (not tested):

````
location / {
    content_by_lua '
        ledge:bind("before_request", function(params) 
             params.headers.host = "overridenhost.com"
        end)
        ledge:run()
    ';
}
````

**Comment by keen99** [comment]
+1 on having to patch ledge.lua to ```headers['host'] = self:config_get("upstream_host")```  for the host header to be set "correctly" for my upstream.

**Comment by pintsized** [comment]
Fixed in #99
